qutebrowser/browser/network/networkmanager.py

```diff
@@ -337,6 +337,18 @@ def createRequest(self, op, req, outgoing_data):
             dnt = '0'.encode('ascii')
         req.setRawHeader('DNT'.encode('ascii'), dnt)
         req.setRawHeader('X-Do-Not-Track'.encode('ascii'), dnt)
+
+        current_url = objreg.get('tabbed-browser', scope='window',
+                                 window=self._win_id).currentWidget().url()
```

Comment:
Using `currentWidget()` here looks like it'd do the wrong thing - after all, network requests also can happen in tabs other than the current one :wink:

This should probably be something like `objreg.get('webview', scope='tab', window=self._win_id, tab=self._tab_id).url()` instead (untested)